brownie/project/main.py

```diff
@@ -409,9 +417,21 @@ def check_for_project(path: Union[Path, str] = ".") -> Optional[Path]:
     """Checks for a Brownie project."""
     path = Path(path).resolve()
     for folder in [path] + list(path.parents):
-        if next((i for i in folder.glob("contracts/**/*") if i.suffix in (".vy", ".sol")), None):
+
+        structure_config = _load_project_structure_config(folder)
+        contracts_path = structure_config["contracts"]
+        tests_path = structure_config["tests"]
+
+        if next(
+            (
+                i
+                for i in folder.joinpath(contracts_path).glob("**/*")
+                if i.suffix in (".vy", ".sol")
+            ),
+            None,
+        ):
             return folder
-        if folder.joinpath("contracts").is_dir() and folder.joinpath("tests").is_dir():
+        if folder.joinpath(tests_path).is_dir() and folder.joinpath(tests_path).is_dir():
```

Comment:
The first check should be `contracts_path`

Reply:
Fixed it, and refactored a bit to make that `next` statement less ugly.  I hadn't seen what `black` did to it :eyes: